### ISSUE TITLE
508-defect-2 [AXE-CORE, SEMANTIC MARKUP] Missing <dl> wrapper in FileField

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#9088]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

This is a problem occurs when a page includes both a widget (e.g. yesNo widget) and a file upload (using `fileUploadUI`) component, e.g form 526 review and submit page > "Supporting evidence" accordion > "Private medical records" page 

<details><summary>axe-coconut error</summary>

![Screen Shot 2020-05-13 at 10 38 53 AM](https://user-images.githubusercontent.com/136959/81874955-c003a300-9544-11ea-8e50-05048c3f3f7b.png)
</details>

### Non axe-error situation

When a page only includes the file upload component, e.g., form 526 review and submit page > "Supporting evidence" accordion > "Lay statements and other evidence", the "review" wrapper is set as a `div`; this is because:

- The review page [`ObjectField` checks for the `addAnotherLabel` option](https://github.com/department-of-veterans-affairs/vets-website/pull/12449/files#diff-0bef2cfd7e8a1088ee9e267ff7d8eaceR135)
- Then to compensate, the [`FileField` moves the `dl`](https://github.com/department-of-veterans-affairs/vets-website/pull/12449/files#diff-3032c07cd04958c35864be081380a546R120) so it immediately wraps the "review-row"

This works great! In this case.

<details><summary>"Lay statements and other evidence" page screenshot</summary>

![Screen Shot 2020-05-13 at 4 28 49 PM](https://user-images.githubusercontent.com/136959/81867913-0fdb6d80-9537-11ea-97d4-5690c982a58b.png)</details>

### Axe error situation

With the addition of another component, a `yesNo` widget in this case:

- The `ObjectField` detects the `addAnotherLabel` option, and sets to wrap the content in a `div`
- The `yesNo` widget
  - It is unable to detect that it's inside the wrapping `ObjectField`
  - Within the `ObjectField` it uses `SchemaField` (`ReviewFieldTemplate`) to render the yesNo widget in a `<div class="review-row">` wrapping the  `<dt>` and `<dd>`.
- The `FileField` moves the `dl` to wrap the "review-row"

<details><summary>This results in this HTML</summary>

```html
<div class="review">
  <!-- yesNo widget -->
  <div class="review-row">
    <dt>...</dt>
    <dd>...</dd>
  </div>
  <!-- FileField -->
  <div class="schemaform-file-upload-review">
    <ul class="schemaform-file-list">
      <li id="..." class="va-growable-background">
        <p> </p>
        <span><strong>File1.png</strong></span>
        <dl class="schemaform-file-attachment review">
          <div class="review-row">
            <dt>Document type</dt>
            <dd><span>Other</span></dd>
          </div>
        </dl>
        <div><button>Delete file</button></div>
      </li>
    </ul>
  </div>
</div>
```
</details>

<details><summary>"Private medical records" page screenshot</summary>

![Screen Shot 2020-05-13 at 6 02 28 PM](https://user-images.githubusercontent.com/136959/81874738-26d48c80-9544-11ea-8db2-879145752215.png)</details>

## Acceptance Criteria

- Fix axe error

## Environment

* Operating System: All
* Browser: All
* Screenreading device: N/A
* Server destination: All

## Steps to Recreate

1. Log in to user 228
1. Start form 526, and work your way to this page: https://staging.va.gov/disability/file-disability-claim-form-21-526ez/supporting-evidence/evidence-types
1. Choose "Yes" to the "Is there evidence" question
1. Select checkboxes: "Privated medical records" & "Supporting (lay)..."
1. Continue in the form until you reach the review & submit page
1. Open the "Supporting evidence" accordion
1. Run axe-coconut

## Possible Fixes (optional)

In [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/12654/commits/c57e370dc0cb495998d7e91ed47b882cdcd968be), I attempted to fix this problem by:

- Adding a `useDlWrap: true` flag in the uiSchema
- In the `src/platform/forms-system/src/js/review/ReviewFieldTemplate.jsx` file, the uiSchema option was checked, and if set, use a `dl`.
- A lot of CSS was also needed to visually fix the component since formation defines a `dl.review`, but we need a `div.review` in this situration

## WCAG or Vendor Guidance (optional)

https://dequeuniversity.com/rules/axe/3.5/dlitem?application=coconut

## Screenshots or Trace Logs

See above

**Comment by davidakennedy** [comment]
I retested this ticket with the help of @Mottie. It's no longer an issue, and was addressed in a past pull request. We can close this one.

**Comment by taylorkaren** [comment]
Closing this ticket as it is no longer an issue.
